clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -859,6 +862,9 @@ public SenderMetrics(SenderMetricsRegistry metrics, Metadata metadata, KafkaClie
             this.recordsPerRequestSensor.add(new Meter(metrics.recordSendRate, metrics.recordSendTotal));
             this.recordsPerRequestSensor.add(metrics.recordsPerRequestAvg, new Avg());
 
+            this.metadataRequestRateSensor = metrics.sensor("metadata-request-rate");
```

Comment:
Again, since the KafkaProducer and the Sender share the same ProducerMetadata object.
Can we move the sensor inside the ProducerMetadata and consolidate them into one single sensor instead of two?

Reply:
@gitlw , similar to the reason I mentioned in other comment. I think we may not be able to include the sensor into ProducerMetadata class.